### PR TITLE
fixing TA-968 to persist the messages in ActivemQ embedded broker

### PR DESCRIPTION
**Pull request by nuwanw**

### CODE REVIEW

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSQueueMessageProducer.java [hunk 1]
```diff
@@ -70,6 +70,31 @@ public void connect(String queueName) throws JMSException, NamingException {
     }
 
     /**
+     * This will establish  the connection with the given Queue.
+     * This must be called before calling pushMessage() to send messages
+     *
+     * @param queueName name of the Queue
+     * @throws JMSException
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSQueueMessageProducer.java]: Explain why these exceptions are thrown

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSTopicMessagePublisher.java [hunk 2]
```diff
@@ -72,6 +72,31 @@ public void connect(String topicName) throws JMSException, NamingException {
     }
 
     /**
+     * This will establish  the connection with the given Topic.
+     * This must be called before calling publish()
+     *
+     * @param topicName name of the topic
+     * @throws JMSException
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSTopicMessagePublisher.java]: explain the exceptions

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSQueueMessageProducer.java (hunk 1 above)
Comment (nuwanw) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSQueueMessageProducer.java]: Done

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSTopicMessagePublisher.java (hunk 2 above)
Comment (nuwanw) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/jmsserver/client/JMSTopicMessagePublisher.java]: Done

**Pull request merged**